addons/project/tests/test_project_recurrence.py

```diff
@@ -113,3 +113,46 @@ def test_recurring_settings_change(self):
             .create({'group_project_recurring_tasks': False}) \
             .execute()
         self.assertFalse(test_task.recurring_task, 'The "Recurring" feature should not be enabled by default.')
+
+    def test_recurrence_disabled_with_single_task(self):
+        config_settings = self.env['res.config.settings'].create({'group_project_recurring_tasks': True})
+        config_settings.execute()
+        with freeze_time(self.date_01_01):
+            form = Form(self.env['project.task'])
+            form.name = 'recurring task'
+            form.project_id = self.project_recurring
+            form.recurring_task = True
+            form.repeat_interval = 5
+            form.repeat_unit = 'month'
+            form.repeat_type = 'forever'
+            task = form.save()
+        self.assertTrue(task.recurrence_id, 'Should create a recurrence')
+        config_settings = self.env['res.config.settings'].create({'group_project_recurring_tasks': False})
+        config_settings.execute()
+        self.assertFalse(task.recurring_task, 'Recurrence should be disabled and Recurrence button should not be visible for task')
+
+    def test_recurrence_disabled_multi_tasks(self):
+        config_settings = self.env['res.config.settings'].create({'group_project_recurring_tasks': True})
+        config_settings.execute()
+        with freeze_time(self.date_01_01):
+            form = Form(self.env['project.task'])
+            form.name = 'recurring task'
+            form.project_id = self.project_recurring
+            form.recurring_task = True
+            form.repeat_interval = 5
+            form.repeat_unit = 'month'
+            form.repeat_type = 'forever'
+            task = form.save()
+        with freeze_time(self.date_01_01 + relativedelta(months=1)):
+            task.state = '1_done'
+            form.save()
+        task2 = task.recurrence_id.task_ids.filtered(lambda t: t.id != task.id)
+        config_settings = self.env['res.config.settings'].create({'group_project_recurring_tasks': False})
```

Comment:
Rather than disabling the recurrence option, we should test to set `recurring_task` to False on task2.
Am I missing the point?

Reply:
hello @auon-odoo ,
Changes are done